scripts/components/sparse_test_matrix_generator.ts

```diff
@@ -0,0 +1,93 @@
+import { glob } from 'glob';
+import path from 'path';
+
+// See https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow
+type JobMatrix = {
+  include?: Array<Record<string, string>>;
+} & Record<string, string>;
+
+export type SparseTestMatrixGeneratorProps = {
+  testGlobPattern: string;
+  maxTestsPerJob: number;
```

Comment:
The ability to have multiple tests per job seems inevitable (GH has 256 limit of jobs per workflow), therefore I added this concept eagerly.